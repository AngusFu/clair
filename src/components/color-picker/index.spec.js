import { mount, createLocalVue } from '@vue/test-utils'
import ColorPicker from './index.vue'

// TODO: resolve issue with $clair
const localVue = createLocalVue()
localVue.prototype.$clair = {
  defaultThrottleTime: 150
}

// TODO: further work needed
it('should render ColorPicker correctly', () => {
  const wrapper = mount(ColorPicker, {
    localVue,
    propsData: {
      mode: 'rgb',
      value: '#2f85da'
    }
  })
  const colorPicker = wrapper.vm.$refs.panel
  expect(wrapper).toMatchSnapshot()
  // show portal
  wrapper.vm.$refs.trigger.click()
  expect(colorPicker.$el.innerHTML).toMatchSnapshot()
})

it('should render inline ColorPicker correctly', () => {
  const wrapper = mount(ColorPicker, {
    localVue,
    propsData: {
      mode: 'rgb',
      value: '#ff7b00',
      inline: true
    }
  })

  expect(wrapper).toMatchSnapshot()
})
