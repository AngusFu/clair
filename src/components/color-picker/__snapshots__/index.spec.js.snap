// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ColorPicker correctly 1`] = `
<div class="color-picker__wrapper">
  <!---->
  <div class="color-picker__trigger" style="width: 24px; height: 24px; border-color: transparent;"></div>
</div>
`;

exports[`should render ColorPicker correctly 2`] = `
<div class="cp__v-ctrl cp__saturation">
  <div class="msk-hue" style="background-color: rgb(0, 128, 255);"></div>
  <div class="msk-white"></div>
  <div class="msk-black"></div>
  <p class="cp__thumb" style="left: 78.4%; top: 14.5%;"></p>
</div>
<div class="cp__ctrl-pane">
  <div>
    <div class="cp__preview">
      <div style="background-color: rgb(47, 133, 218); border-color: transparent;"></div>
    </div>
    <div class="cp__tracks">
      <div class="cp__v-ctrl cp__ctrl-bar cp__ctrl-hue">
        <div class="cp__thumb" style="left: 41.7%;"></div>
      </div>
      <div class="cp__v-ctrl cp__ctrl-alpha">
        <div class="cp__thumb" style="left: 100%;"></div>
        <div class="cp__ctrl-bar"></div>
      </div>
    </div>
  </div>
  <div style="margin-top: 10px;">
    <div class="cp__fm-fields">
      <div>
        <div style="position: relative;"><input type="string" maxlength="9"> <span>hex</span></div>
      </div>
    </div>
    <div class="cp__fm-switcher">
      <div><svg viewBox="0 0 24 24">
          <path fill="#333" d="M12,5.83L15.17,9L16.58,7.59L12,3L7.41,7.59L8.83,9L12,5.83Z"></path>
          <path fill="#333" d="M12,18.17L8.83,15L7.42,16.41L12,21L16.59,16.41L15.17,15Z"></path>
        </svg></div>
    </div>
  </div>
</div>
`;

exports[`should render inline ColorPicker correctly 1`] = `
<div class="cp__wrapper">
  <div class="cp__v-ctrl cp__saturation">
    <div class="msk-hue" style="background-color: rgb(255, 123, 0);"></div>
    <div class="msk-white"></div>
    <div class="msk-black"></div>
    <p class="cp__thumb" style="left: 100%; top: 0.00%;"></p>
  </div>
  <div class="cp__ctrl-pane">
    <div>
      <div class="cp__preview">
        <div style="background-color: rgb(255, 123, 0); border-color: transparent;"></div>
      </div>
      <div class="cp__tracks">
        <div class="cp__v-ctrl cp__ctrl-bar cp__ctrl-hue">
          <div class="cp__thumb" style="left: 92.0%;"></div>
        </div>
        <div class="cp__v-ctrl cp__ctrl-alpha">
          <div class="cp__thumb" style="left: 100%;"></div>
          <div class="cp__ctrl-bar"></div>
        </div>
      </div>
    </div>
    <div style="margin-top: 10px;">
      <div class="cp__fm-fields">
        <div>
          <div style="position: relative;"><input type="string" maxlength="9"> <span>hex</span></div>
        </div>
      </div>
      <div class="cp__fm-switcher">
        <div><svg viewBox="0 0 24 24">
            <path fill="#333" d="M12,5.83L15.17,9L16.58,7.59L12,3L7.41,7.59L8.83,9L12,5.83Z"></path>
            <path fill="#333" d="M12,18.17L8.83,15L7.42,16.41L12,21L16.59,16.41L15.17,15Z"></path>
          </svg></div>
      </div>
    </div>
  </div>
</div>
`;
